subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/SkipUpToDateStepTest.groovy

```diff
@@ -16,20 +16,21 @@
 
 package org.gradle.internal.execution.steps
 
+import com.google.common.collect.ImmutableSortedMap
+import org.gradle.caching.internal.origin.OriginMetadata
+import org.gradle.internal.Try
 import org.gradle.internal.change.ChangeVisitor
 import org.gradle.internal.change.DescriptiveChange
 import org.gradle.internal.execution.ExecutionOutcome
 import org.gradle.internal.execution.IncrementalChangesContext
-import org.gradle.internal.execution.Step
-import org.gradle.internal.execution.UnitOfWork
+import org.gradle.internal.execution.SnapshotResult
 import org.gradle.internal.execution.history.changes.ExecutionStateChanges
-import org.gradle.testing.internal.util.Specification
+import org.gradle.internal.fingerprint.impl.EmptyCurrentFileCollectionFingerprint
 
-class SkipUpToDateStepTest extends Specification {
-    def delegate = Mock(Step)
+class SkipUpToDateStepTest extends StepSpec {
     def step = new SkipUpToDateStep<IncrementalChangesContext>(delegate)
     def context = Mock(IncrementalChangesContext)
-    def work = Mock(UnitOfWork)
+
```

Comment:
Nit: extra new line.

Reply:
These are intentional. In the tests I tried to group things that belong to the pipeline architecture `step`, `context` and `delegateResult` together. Also, services are in a separate group wherever appropriate.